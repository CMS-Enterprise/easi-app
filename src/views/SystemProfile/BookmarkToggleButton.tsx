import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useMutation } from '@apollo/client';
import { Button, IconBookmark } from '@trussworks/react-uswds';
import classnames from 'classnames';

import CreateCedarSystemBookmarkQuery from 'queries/CreateCedarSystemBookmarkQuery';
import DeleteCedarSystemBookmarkQuery from 'queries/DeleteCedarSystemBookmarkQuery';
import {
  CreateCedarSystemBookmark,
  CreateCedarSystemBookmarkVariables
} from 'queries/types/CreateCedarSystemBookmark';
import {
  DeleteCedarSystemBookmark,
  DeleteCedarSystemBookmarkVariables
} from 'queries/types/DeleteCedarSystemBookmark';

export default function BookmarkToggleButton({
  id,
  initialBookmarked
}: {
  id: string;
  initialBookmarked: boolean;
}) {
  const { t } = useTranslation('systemProfile');

  const [isBookmarked, setBookmarked] = useState<boolean>(initialBookmarked);

  const [create, { loading: createLoading }] = useMutation<
    CreateCedarSystemBookmark,
    CreateCedarSystemBookmarkVariables
  >(CreateCedarSystemBookmarkQuery);

  const [del, { loading: delLoading }] = useMutation<
    DeleteCedarSystemBookmark,
    DeleteCedarSystemBookmarkVariables
  >(DeleteCedarSystemBookmarkQuery);

  const toggle = () => {
    if (createLoading || delLoading) return;

    (isBookmarked ? del : create)({
      variables: {
        input: {
          cedarSystemId: id
        }
      }
    }).then(res => {
      setBookmarked(!isBookmarked);
    });
  };

  return (
    <Button
      type="button"
      className="usa-button display-flex flex-align-center margin-right-0 padding-y-1 padding-left-105 padding-right-2 radius-0 bg-white text-primary"
      onClick={toggle}
    >
      <IconBookmark
        size={3}
        className={classnames(
          'margin-right-1',
          isBookmarked ? 'text-primary' : 'text-base-lighter'
        )}
      />
      {t(`bookmark.${isBookmarked ? 'bookmarked' : 'bookmark'}`)}
    </Button>
  );
}
