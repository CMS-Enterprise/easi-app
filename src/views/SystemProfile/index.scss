#system-profile {
    .system-profile {
        &__bookmark {
            font-size: 1.5rem;
            padding-top: .75rem;
            cursor: initial;
        }

        &__main-nav-link {
            color: color($theme-color-primary);
            margin-left: -0.5rem;
            margin-top: -5rem;
            margin-bottom: 2.5rem;
        }
    }

  .description-truncated {
    .description-definition {
      display: -webkit-box;
      -webkit-line-clamp: 3;
      -webkit-box-orient: vertical;
      overflow: hidden;
    }
    &.expanded {
      .description-definition {
        display: block;
        overflow: auto;
      }
      .expand-icon {
        transform: rotate(180deg);
      }
    }
  }

    .sticky-nav {
        align-self: start;
        position: sticky;
        position: -webkit-sticky;
        top: 0;
        padding-top: 1rem;
        @media screen and (min-width: $desktop) {
            margin-top: -1rem;
          }
    }

    .link-header {
        text-underline-offset: 0.25rem;
    }

  .url-card-link {
    display: block;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

    .usa-sidenav, .usa-nav__primary-item {
        border: none;

        &__item {
            border: none;
        }
    }

    .usa-card__container {
        border-radius: .25rem;
        margin: 0;
    }

    .usa-sidenav__item a:not(.usa-current) {
        padding-left: 0;
    }

    .usa-nav__close {
        z-index: 1000;
    }

    .nav-group-border {
        border-bottom: .5px solid color($theme-color-disabled-light);
    }

  dd.description-definition {
    overflow-wrap: break-word;
  }
}

.top-divider {
  @media screen and (min-width: $desktop) {
      height: .25rem;
      width: 100%;
      background-color: color($theme-color-primary-lighter);
  }
}

.side-divider {
  padding-left: 2rem;
  @media screen and (max-width: $desktop) {
      border-left: .25rem solid color($theme-color-primary-lighter);
  }
}
