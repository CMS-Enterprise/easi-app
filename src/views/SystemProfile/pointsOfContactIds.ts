import { RoleTypeId, SubpageKey } from 'types/systemProfile';

/**
 * Points of Contact Cedar Role Type Ids by System Profile subpage.
 * The Contact Role Type Id lists are contextualized by subpage.
 * Lists are in order of priority.
 */
const pointsOfContactIds: Record<SubpageKey, RoleTypeId[]> = {
  home: [RoleTypeId.PROJECT_LEAD],
  details: [
    RoleTypeId.PROJECT_LEAD,
    RoleTypeId.SYSTEM_BUSINESS_QUESTION_CONTACT,
    RoleTypeId.SYSTEM_DATA_CENTER_CONTACT
  ],
  team: [
    RoleTypeId.PROJECT_LEAD,
    RoleTypeId.COR,
    RoleTypeId.SYSTEM_BUSINESS_QUESTION_CONTACT
  ],
  'funding-and-budget': [
    RoleTypeId.COR,
    RoleTypeId.SYSTEM_BUSINESS_QUESTION_CONTACT,
    RoleTypeId.PROJECT_LEAD
  ],
  'tools-and-software': [
    RoleTypeId.SYSTEM_MAINTAINER,
    RoleTypeId.SYSTEM_ISSUES_CONTACT,
    RoleTypeId.PROJECT_LEAD
  ],
  ato: [
    RoleTypeId.ISSO,
    RoleTypeId.SYSTEM_ISSUES_CONTACT,
    RoleTypeId.PROJECT_LEAD
  ],
  'lifecycle-id': [
    RoleTypeId.PROJECT_LEAD,
    RoleTypeId.COR,
    RoleTypeId.SYSTEM_BUSINESS_QUESTION_CONTACT
  ],
  'section-508': [
    RoleTypeId.PROJECT_LEAD,
    RoleTypeId.SYSTEM_MAINTAINER,
    RoleTypeId.SYSTEM_ISSUES_CONTACT
  ],
  'sub-systems': [
    RoleTypeId.PROJECT_LEAD,
    RoleTypeId.SYSTEM_MAINTAINER,
    RoleTypeId.SYSTEM_BUSINESS_QUESTION_CONTACT
  ],
  'system-data': [
    RoleTypeId.SYSTEM_MAINTAINER,
    RoleTypeId.API_CONTACT,
    RoleTypeId.PROJECT_LEAD
  ],
  documents: [
    RoleTypeId.PROJECT_LEAD,
    RoleTypeId.SYSTEM_BUSINESS_QUESTION_CONTACT,
    RoleTypeId.SYSTEM_MAINTAINER
  ]
};

// All lists are followed by `RoleTypeId.BUSINESS_OWNER`
Object.values(pointsOfContactIds).forEach(ids => {
  ids.push(RoleTypeId.BUSINESS_OWNER);
});

export default pointsOfContactIds;
