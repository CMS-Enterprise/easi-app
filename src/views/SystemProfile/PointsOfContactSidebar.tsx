import React from 'react';
import { useTranslation } from 'react-i18next';
import { IconMailOutline, Link } from '@trussworks/react-uswds';

import UswdsReactLink from 'components/LinkWrapper';
import Alert from 'components/shared/Alert';
import {
  SubpageKey,
  SystemProfileSubviewProps,
  UsernameWithRoles
} from 'types/systemProfile';

import { getPersonFullName } from './helpers';
import pointsOfContactIds from './pointsOfContactIds';

/**
 * Get a list of subpage contacts defined by `pointsOfContactIds`.
 * Return all members of the first matching Role Type Id found in the priority list.
 */
export function getPointsOfContact(
  subpageKey: SubpageKey,
  usernamesWithRoles: UsernameWithRoles[]
): UsernameWithRoles[] {
  const subPocIds = pointsOfContactIds[subpageKey];
  let contacts: UsernameWithRoles[] = [];

  // eslint-disable-next-line no-restricted-syntax
  for (const pocid of subPocIds) {
    const found = usernamesWithRoles.filter(user =>
      user.roles.find(r => r.roleTypeName === pocid)
    );
    if (found.length) {
      contacts = found;
      break;
    }
  }

  return contacts;
}

interface PointsOfContactSidebarProps extends SystemProfileSubviewProps {
  subpageKey: SubpageKey;
  systemId: string;
}

const PointsOfContactSidebar = ({
  system,
  subpageKey,
  systemId
}: PointsOfContactSidebarProps) => {
  const { t } = useTranslation('systemProfile');

  const contactsWithRoles = getPointsOfContact(
    subpageKey,
    system.usernamesWithRoles
  );
  return (
    <>
      <p className="font-body-xs margin-top-1 margin-bottom-3">
        {t('singleSystem.pointsOfContact', { count: contactsWithRoles.length })}
      </p>
      {contactsWithRoles.length ? (
        contactsWithRoles.map(contact => {
          const role = contact.roles[0];
          return (
            <React.Fragment key={role.roleID}>
              <h3 className="system-profile__subheader margin-bottom-1">
                {getPersonFullName(role)}
              </h3>
              <div>
                {contact.roles.map(r => (
                  <h5
                    key={r.roleID}
                    className="margin-top-0 margin-bottom-05 font-sans-2xs text-normal"
                  >
                    {r.roleTypeName}
                  </h5>
                ))}
              </div>
              {role.assigneeEmail && (
                <p>
                  <Link
                    aria-label={t('singleSystem.sendEmail')}
                    className="line-height-body-5"
                    href={`mailto:${role.assigneeEmail}`}
                    target="_blank"
                  >
                    {t('singleSystem.sendEmail')}
                    <IconMailOutline className="margin-left-05 margin-bottom-2px text-tbottom" />
                  </Link>
                </p>
              )}
            </React.Fragment>
          );
        })
      ) : (
        // No contacts
        <Alert type="info" slim className="margin-bottom-6">
          {t(`singleSystem.noPointsOfContact`)}
        </Alert>
      )}
      {subpageKey !== 'team' && (
        <p>
          <UswdsReactLink
            aria-label={t('singleSystem.moreContact')}
            className="line-height-body-5"
            to={`/systems/${systemId}/team`}
          >
            {t('singleSystem.moreContact')}
            <span aria-hidden>&nbsp;</span>
            <span aria-hidden>&rarr; </span>
          </UswdsReactLink>
        </p>
      )}
    </>
  );
};

export default PointsOfContactSidebar;
