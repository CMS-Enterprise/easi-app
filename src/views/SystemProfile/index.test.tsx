import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import {
  fireEvent,
  render,
  waitFor,
  waitForElementToBeRemoved
} from '@testing-library/react';
import { clone, cloneDeep } from 'lodash';
import { DateTime } from 'luxon';

import { ATO_STATUS_DUE_SOON_DAYS } from 'constants/systemProfile';
import {
  getMockPersonRole,
  getMockSystemProfileData,
  query,
  result
} from 'data/mock/systemProfile';
// eslint-disable-next-line camelcase
import { GetSystemProfile_cedarSystemDetails_roles } from 'queries/types/GetSystemProfile';
import { RoleTypeId, SubpageKey } from 'types/systemProfile';

import SystemProfile, { getAtoStatus } from './index';
import pointsOfContactIds from './pointsOfContactIds';
import PointsOfContactSidebar, {
  getPointsOfContact
} from './PointsOfContactSidebar';

describe('System Profile parent request', () => {
  it('matches snapshot', async () => {
    const { asFragment, getByText, getAllByText, getByTestId } = render(
      <MemoryRouter initialEntries={['/systems/000-100-0/home']}>
        <Route path="/systems/:systemId/:subinfo">
          <MockedProvider mocks={[query]} addTypename={false}>
            <SystemProfile />
          </MockedProvider>
        </Route>
      </MemoryRouter>
    );
    await waitForElementToBeRemoved(() => getByTestId('page-loading'));
    expect(asFragment()).toMatchSnapshot();
    expect(getByText('CMS.gov')).toBeInTheDocument();
    expect(getAllByText('Jane Doe')[0]).toBeInTheDocument();
  });
});

describe('System profile description is expandable', () => {
  beforeEach(() => {
    // Fill in heights for description text element
    // https://github.com/testing-library/react-testing-library/issues/353
    const el = HTMLElement.prototype;
    Object.defineProperty(el, 'offsetHeight', {
      configurable: true,
      value: 300
    });
    Object.defineProperty(el, 'scrollHeight', {
      configurable: true,
      value: 400
    });
  });

  it('shows read more & less', async () => {
    const { getByText, findByTestId } = render(
      <MemoryRouter initialEntries={['/systems/000-100-0/home']}>
        <Route path="/systems/:systemId/:subinfo">
          <MockedProvider mocks={[query]} addTypename={false}>
            <SystemProfile />
          </MockedProvider>
        </Route>
      </MemoryRouter>
    );

    const loading = await findByTestId('page-loading');
    await waitFor(() => {
      expect(loading).not.toBeInTheDocument();
    });

    const readMore = getByText(/read more/i);
    await waitFor(() => {
      expect(readMore).toBeInTheDocument();
    });

    fireEvent.click(readMore);

    await waitFor(() => {
      expect(getByText(/read less/i)).toBeInTheDocument();
    });
  });
});

describe('System Profile ATO Status', () => {
  it('output "No ATO" on missing cedarAuthorityToOperate', () => {
    expect(getAtoStatus(undefined)).toBe('No ATO');
  });

  it('output "No ATO" on missing cedarAuthorityToOperate.dateAuthorizationMemoExpires', () => {
    const cedarAto = clone(result.data.cedarAuthorityToOperate[0]);

    cedarAto.dateAuthorizationMemoExpires = '';
    expect(getAtoStatus(cedarAto)).toBe('No ATO');

    cedarAto.dateAuthorizationMemoExpires = null;
    expect(getAtoStatus(cedarAto)).toBe('No ATO');
  });

  it.each([
    { status: 'Expired', dt: DateTime.utc().minus({ days: 1 }) },
    {
      status: 'Due Soon',
      dt: DateTime.utc().plus({ days: ATO_STATUS_DUE_SOON_DAYS })
    },
    {
      status: 'Active',
      dt: DateTime.utc().plus({ days: ATO_STATUS_DUE_SOON_DAYS + 1 })
    }
  ])('output based on current date %j', ({ status, dt }) => {
    const cedarAto = clone(result.data.cedarAuthorityToOperate[0]);
    cedarAto.dateAuthorizationMemoExpires = dt.toString();
    expect(getAtoStatus(cedarAto)).toBe(status);
  });
});

describe('System Profile Points of Contact by subpage', () => {
  const resultdata = cloneDeep(result.data);

  // Reassign 1 person per Cedar Role Type
  resultdata.cedarSystemDetails!.roles = Object.values(
    RoleTypeId
    // eslint-disable-next-line camelcase
  ).map<GetSystemProfile_cedarSystemDetails_roles>((roleTypeID, idx) =>
    getMockPersonRole({
      assigneeUsername: `ABC${idx}`,
      roleTypeID
    })
  );

  const data = getMockSystemProfileData(resultdata);

  it.each(
    Object.keys(pointsOfContactIds).map(subpage => subpage as SubpageKey)
  )('poc set is of the first matching role type for %s', subpage => {
    const allowedSubpagePocIds = pointsOfContactIds[subpage];
    const contacts = getPointsOfContact(subpage, data.usernamesWithRoles);

    // Collect all contacts matching all poc ids for the subpage
    const received = allowedSubpagePocIds.filter(pocid => {
      return contacts.every(contact =>
        contact.roles.some(r => r.roleTypeID === pocid)
      );
    });

    // The set of contacts are all of the same role type as allowed in the poc list
    expect(received).toHaveLength(1);
    // It is the first from the priority list
    expect(received[0]).toEqual(allowedSubpagePocIds[0]);
  });

  it('displays alert on no poc memebers', () => {
    const res = cloneDeep(result.data);
    res.cedarSystemDetails!.roles = [];
    const dataWithoutTeam = getMockSystemProfileData(res);
    const { getByTestId } = render(
      <MemoryRouter>
        <PointsOfContactSidebar
          subpageKey="home"
          systemId=""
          system={dataWithoutTeam}
        />
      </MemoryRouter>
    );
    expect(getByTestId('alert')).toBeDefined();
  });
});
