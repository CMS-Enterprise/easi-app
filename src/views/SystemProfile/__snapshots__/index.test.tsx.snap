// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`System Profile parent request > matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="easi-main-content"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <div
      id="system-profile"
    >
      <div
        class="usa-summary-box padding-0 border-0 radius-0 bg-primary-lighter"
        data-testid="summary-box"
      >
        <div
          class="usa-summary-box__body"
        >
          <h3
            class="usa-summary-box__heading"
          />
          <div
            class="usa-summary-box__text"
          >
            <div
              class="padding-top-3 padding-bottom-3 margin-top-neg-1 height-full"
            >
              <div
                class="grid-container"
                data-testid="grid"
              >
                <nav
                  aria-label="Breadcrumbs"
                  class="usa-breadcrumb usa-breadcrumb--wrap bg-transparent padding-0"
                >
                  <ol
                    class="usa-breadcrumb__list"
                  >
                    <li
                      class="usa-breadcrumb__list-item"
                    >
                      <span>
                        ← 
                      </span>
                      <a
                        class="usa-breadcrumb__link"
                        href="/systems"
                      >
                        <span>
                          Back to All systems
                        </span>
                      </a>
                    </li>
                  </ol>
                </nav>
                <h1
                  aria-live="polite"
                  class="easi-h1 margin-top-2"
                  tabindex="-1"
                >
                  <svg
                    class="usa-icon usa-icon--size-4 text-primary"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                   
                  <span>
                    CMS.gov 
                  </span>
                  <span
                    class="text-normal font-body-sm"
                  >
                    (cms.hhs.gov)
                  </span>
                  <div
                    class="text-normal font-body-md"
                  >
                    <div
                      class="easi-collapsable-link"
                    >
                      <div
                        class="easi-collapsable-link__content-no-bar"
                        id="single-system-profile"
                      >
                        <div
                          class="description-truncated margin-bottom-2"
                        >
                          <dd
                            class="description-definition margin-0 font-body-lg line-height-body-5 text-light"
                          >
                            CMS.gov provides the public and professionals the ability to access information regarding the CMS programs. CMS.gov website mission is to provide clear, accurate, and timely information about CMS programs to the entire health care community to improve quality and efficiency in an evolving health care system. CMS.gov website is a combination of static content and general content applications.  Core Function: * Information dissemination
                          </dd>
                        </div>
                        <a
                          aria-label="Open system external link"
                          class="usa-link usa-link--external line-height-body-5"
                          href="https://cms.gov"
                          target="_blank"
                        >
                          View CMS.gov
                          <span
                            aria-hidden="true"
                          >
                             
                          </span>
                        </a>
                        <div
                          class="grid-row margin-top-3"
                          data-testid="grid"
                        >
                          <div
                            class="desktop:grid-col-6 margin-bottom-2"
                            data-testid="grid"
                          >
                            <dd
                              class="description-definition margin-0"
                            >
                              CMS Component
                            </dd>
                            <dt
                              class="text-bold margin-bottom-05 font-body-md"
                            >
                              Web and Emerging Technologies Group
                            </dt>
                          </div>
                          <div
                            class="desktop:grid-col-6 margin-bottom-2"
                            data-testid="grid"
                          >
                            <dd
                              class="description-definition margin-0"
                            >
                              Business Owner
                            </dd>
                            <dt
                              class="text-bold margin-bottom-05 font-body-md"
                            >
                              Audrey Abrams
                            </dt>
                          </div>
                        </div>
                      </div>
                      <button
                        aria-controls="single-system-profile"
                        aria-expanded="true"
                        class="usa-button usa-button--unstyled display-flex flex-align-center text-bold margin-top-3"
                        data-testid="collapsable-link"
                        type="button"
                      >
                        <svg
                          class="usa-icon easi-collapsable-link__eye-icon"
                          focusable="false"
                          height="1em"
                          role="img"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                            fill="none"
                          />
                          <path
                            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46A11.804 11.804 0 0 0 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                          />
                        </svg>
                        Hide system summary
                      </button>
                    </div>
                  </div>
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="system-detail__subNav-accordion"
      >
        <button
          aria-controls="system-detail__subNav"
          aria-expanded="false"
          class="usa-menu-btn easi-header__basic width-full"
          type="button"
        >
          <h3
            class="padding-left-1"
          >
            System Home
          </h3>
          <svg
            class="usa-icon usa-icon--size-3"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </button>
      </div>
      <div
        class="easi-section margin-bottom-5"
        data-testid="section-wrapper"
      >
        <div
          class="grid-container"
          data-testid="gridContainer"
        >
          <div
            class="grid-row grid-gap"
            data-testid="grid"
          >
            <div
              class="desktop:grid-col-3 padding-right-4 sticky side-nav"
              data-testid="grid"
            >
              <ul
                class="usa-sidenav"
                data-testid="sidenav"
              >
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    aria-current="page"
                    class="nav-group-border usa-current"
                    href="/systems/000-100-0/home"
                  >
                    System Home
                  </a>
                </li>
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    class=""
                    href="/systems/000-100-0/details"
                  >
                    Implementation details
                  </a>
                </li>
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    class="nav-group-border"
                    href="/systems/000-100-0/team"
                  >
                    Team
                  </a>
                </li>
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    class=""
                    href="/systems/000-100-0/ato"
                  >
                    ATO
                  </a>
                </li>
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    class=""
                    href="/systems/000-100-0/funding-and-budget"
                  >
                    Funding and Budget
                  </a>
                </li>
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    class=""
                    href="/systems/000-100-0/tools-and-software"
                  >
                    Tools and Software
                  </a>
                </li>
                <li
                  class="usa-sidenav__item"
                >
                  <a
                    class="nav-group-border"
                    href="/systems/000-100-0/sub-systems"
                  >
                    Sub-systems
                  </a>
                </li>
              </ul>
            </div>
            <div
              class="desktop:grid-col-9"
              data-testid="grid"
            >
              <div
                id="system-detail"
              >
                <div
                  class="grid-container padding-left-0 padding-right-0"
                  data-testid="gridContainer"
                >
                  <div
                    class="grid-row grid-gap"
                    data-testid="grid"
                  >
                    <div
                      class="desktop:grid-col-8 padding-top-3"
                      data-testid="grid"
                    >
                      <div
                        class="easi-section padding-bottom-4 margin-bottom-4"
                        data-testid="section-wrapper"
                      >
                        <ul
                          class="usa-card-group margin-0"
                          data-testid="CardGroup"
                        >
                          <li
                            class="usa-card grid-col-12"
                            data-testid="Card"
                          >
                            <div
                              class="usa-card__container"
                            >
                              <header
                                class="usa-card__header easi-header__basic padding-2 padding-bottom-0 text-top"
                                data-testid="CardHeader"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-12 padding-0"
                                    data-testid="grid"
                                  >
                                    <dt>
                                      Production environment
                                    </dt>
                                  </div>
                                </div>
                              </header>
                              <div
                                class="usa-card__body padding-x-2 padding-y-0"
                                data-testid="CardBody"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-12 padding-0"
                                    data-testid="grid"
                                  >
                                    <h3
                                      class="link-header margin-top-0 margin-bottom-2"
                                    >
                                      <a
                                        class="usa-link usa-link--external link-header url-card-link"
                                        href="https://cms.gov"
                                        target="_blank"
                                      >
                                        https://cms.gov
                                      </a>
                                    </h3>
                                    <div
                                      class="margin-bottom-2"
                                    >
                                      <a
                                        class="usa-link link-header"
                                        href="/systems/000-100-0/details"
                                      >
                                        View URLs
                                        <span
                                          aria-hidden="true"
                                        >
                                           
                                        </span>
                                        <span
                                          aria-hidden="true"
                                        >
                                          → 
                                        </span>
                                      </a>
                                    </div>
                                  </div>
                                </div>
                                <div
                                  class="easi-divider"
                                />
                              </div>
                              <div
                                class="usa-card__footer padding-0"
                                data-testid="CardFooter"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="padding-2"
                                    data-testid="grid"
                                  >
                                    <dt
                                      class="text-bold margin-bottom-05"
                                    >
                                      Location or cloud service provider name
                                    </dt>
                                    <dd
                                      class="description-definition margin-0 line-height-body-3"
                                    >
                                      AWS US East/West
                                    </dd>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </li>
                          <li
                            class="usa-card grid-col-12"
                            data-testid="Card"
                          >
                            <div
                              class="usa-card__container"
                            >
                              <header
                                class="usa-card__header easi-header__basic padding-2 padding-bottom-0 text-top"
                                data-testid="CardHeader"
                              >
                                <dt>
                                  ATO Expiration
                                </dt>
                                <div
                                  class="text-right margin-bottom-0"
                                >
                                  <span
                                    class="easi-tag grid-col-12 bg-error-dark text-white"
                                    data-testid="tag"
                                  >
                                    Expired
                                  </span>
                                </div>
                              </header>
                              <div
                                class="usa-card__body padding-x-2 padding-y-0"
                                data-testid="CardBody"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-12 padding-0"
                                    data-testid="grid"
                                  >
                                    <h3
                                      class="link-header margin-top-0 margin-bottom-2"
                                    >
                                      July 30, 2022
                                    </h3>
                                    <div
                                      class="margin-bottom-2"
                                    >
                                      <a
                                        class="usa-link link-header"
                                        href="/systems/000-100-0/ato"
                                      >
                                        View ATO information
                                        <span
                                          aria-hidden="true"
                                        >
                                           
                                        </span>
                                        <span
                                          aria-hidden="true"
                                        >
                                          → 
                                        </span>
                                      </a>
                                    </div>
                                  </div>
                                </div>
                                <div
                                  class="easi-divider"
                                />
                              </div>
                              <div
                                class="usa-card__footer padding-0"
                                data-testid="CardFooter"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-6 padding-2"
                                    data-testid="grid"
                                  >
                                    <dt
                                      class="text-bold margin-bottom-05"
                                    >
                                      Total POAMs
                                    </dt>
                                    <dd
                                      class="description-definition margin-0 line-height-body-3"
                                    >
                                      7
                                    </dd>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </li>
                          <li
                            class="usa-card grid-col-12"
                            data-testid="Card"
                          >
                            <div
                              class="usa-card__container"
                            >
                              <header
                                class="usa-card__header easi-header__basic padding-2 padding-bottom-0 text-top"
                                data-testid="CardHeader"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-12 padding-0"
                                    data-testid="grid"
                                  >
                                    <dt>
                                      Total Full Time Employees (FTE)
                                    </dt>
                                  </div>
                                </div>
                              </header>
                              <div
                                class="usa-card__body padding-x-2 padding-y-0"
                                data-testid="CardBody"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-12 padding-0"
                                    data-testid="grid"
                                  >
                                    <h3
                                      class="link-header margin-top-0 margin-bottom-2"
                                    >
                                      24
                                    </h3>
                                    <a
                                      class="usa-link link-header"
                                      href="/systems/000-100-0/team"
                                    >
                                      View more Team and Contract information
                                      <span
                                        aria-hidden="true"
                                      >
                                         
                                      </span>
                                      <span
                                        aria-hidden="true"
                                      >
                                        → 
                                      </span>
                                    </a>
                                  </div>
                                </div>
                                <div
                                  class="easi-divider margin-top-2"
                                />
                              </div>
                              <div
                                class="usa-card__footer padding-0"
                                data-testid="CardFooter"
                              >
                                <div
                                  class="grid-row"
                                  data-testid="grid"
                                >
                                  <div
                                    class="desktop:grid-col-6 padding-2"
                                    data-testid="grid"
                                  >
                                    <dt
                                      class="text-bold margin-bottom-05"
                                    >
                                      Federal FTE
                                    </dt>
                                    <dd
                                      class="description-definition margin-0 line-height-body-3"
                                    >
                                      4
                                    </dd>
                                  </div>
                                  <div
                                    class="desktop:grid-col-6 padding-2"
                                    data-testid="grid"
                                  >
                                    <dt
                                      class="text-bold margin-bottom-05"
                                    >
                                      Contractor FTE
                                    </dt>
                                    <dd
                                      class="description-definition margin-0 line-height-body-3"
                                    >
                                      20
                                    </dd>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </li>
                        </ul>
                      </div>
                    </div>
                    <div
                      class="desktop:grid-col-4 sticky side-nav padding-top-7"
                      data-testid="grid"
                    >
                      <div
                        class="side-divider"
                      >
                        <div
                          class="top-divider"
                        />
                        <p
                          class="font-body-xs margin-top-1 margin-bottom-3"
                        >
                          Point of contact
                        </p>
                        <h3
                          class="system-profile__subheader margin-bottom-1"
                        >
                          Ally Anderson
                        </h3>
                        <div>
                          <h5
                            class="margin-top-0 margin-bottom-05 font-sans-2xs text-normal"
                          >
                            Project Lead
                          </h5>
                          <h5
                            class="margin-top-0 margin-bottom-05 font-sans-2xs text-normal"
                          >
                            System Issues Contact
                          </h5>
                          <h5
                            class="margin-top-0 margin-bottom-05 font-sans-2xs text-normal"
                          >
                            QA Reviewer
                          </h5>
                        </div>
                        <p>
                          <a
                            aria-label="Send an email"
                            class="usa-link line-height-body-5"
                            href="mailto:ally.anderson@local.fake"
                            target="_blank"
                          >
                            Send an email
                            <svg
                              class="usa-icon margin-left-05 margin-bottom-2px text-tbottom"
                              focusable="false"
                              height="1em"
                              role="img"
                              viewBox="0 0 24 24"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M0 0h24v24H0z"
                                fill="none"
                              />
                              <path
                                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
                              />
                            </svg>
                          </a>
                        </p>
                        <p>
                          <a
                            class="usa-link line-height-body-5"
                            href="/systems/000-100-0/team"
                          >
                            More points of contact
                            <span
                              aria-hidden="true"
                            >
                               
                            </span>
                            <span
                              aria-hidden="true"
                            >
                              → 
                            </span>
                          </a>
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
